src/types/state.js

```diff
@@ -51,7 +51,27 @@ export type RightClickedMarker = {|
   +markerIndex: MarkerIndex,
 |};
 
-export type ProfileViewState = {|
+/**
+ * Full profile view state
+ * They should not be used from the active tab view.
+ * NOTE: This state is empty for now, but will be used later, do not remove.
+ * globalTracks and localTracksByPid states will be here in the future.
+ */
+export type FullProfileViewState = {||};
+
+/**
+ * Active tab profile view state
+ * They should not be used from the full view.
+ */
+export type ActiveTabProfileViewState = {|
+  hiddenGlobalTracksGetter: () => Set<TrackIndex>,
```

Comment:
Yeah, I initially made these read-only. But flow doesn't allow you to make the inner state object properties read-only. Because their containing objects are already read-only, for example here: 
https://github.com/firefox-devtools/profiler/pull/2478/files#diff-1b81febf503e61202f3b62fe7fb7d4f7R93
Also some other inner state objects are also not read-only because of that. For example:
https://github.com/firefox-devtools/profiler/blob/b3efbcaf2203bf2df2fdae348a6ddb751baad3ec/src/types/state.js#L147-L153

If I change these properties into read-only, flow starts to complain about the `combineReducers`.